### PR TITLE
Convert to using react hooks in wallet

### PR DESCRIPTION
**Pull request by Schmavery**
This is very hacky and reimplements parts of reason-apollo, should be revamped when hooks launches for real.
pls don't look at what we do to get the ppx

### CODE REVIEW

File: frontend/wallet/package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "bs-electron": "github:Schmavery/bs-electron",
     "graphql_ppx": "^0.2.8",
     "reason-apollo": "^0.15.2",
-    "reason-react": "^0.5.3"
+    "reason-react": "github:reasonml/reason-react"
```
Comment (bkase) [inline frontend/wallet/package.json]: pin it to a commit (but remember to bump it?)

File: frontend/wallet/postinstall.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+cd ./react_ppx_3/
+opam switch create 4.02.3+buckle-master
```
Comment (bkase) [inline frontend/wallet/postinstall.sh]: remove this line or make sure it no-ops

File: frontend/wallet/react_ppx_3/ppx_source.ml [hunk 3]
```diff
@@ -0,0 +1,1009 @@
+(*
```
Comment (bkase) [inline frontend/wallet/react_ppx_3/ppx_source.ml]: put a comment saying "From foo.com/whatever"

File: frontend/wallet/package.json (hunk 1 above)
Comment (Schmavery) [reply frontend/wallet/package.json]: actually master is broken right now, that's why CI is failing lol

**Comment by Schmavery** [comment]
@bkase darnit this doesn't work because build-wallet doesn't have the opam switch. How against checking in a precompiled mac exec are you (bs-platform includes one in its npm package)

**Pull request merged**